databuilder/extractor/bigquery_usage_extractor.py

```diff
@@ -0,0 +1,103 @@
+from collections import namedtuple
+from datetime import date, timedelta
+import logging
+
+from googleapiclient.discovery import build
+from oauth2client.client import GoogleCredentials
+
+from databuilder.extractor.base_extractor import Extractor
+
+TableColumnUsageTuple = namedtuple('TableColumnUsageTuple', ['database', 'cluster', 'schema',
+                                                             'table', 'column', 'email'])
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BigQueryUsageExtractor(Extractor):
+    """
+    An aggregate extractor for bigquery table usage. This class takes the data from
+    the stackdriver logging API by filtering on timestamp, bigquery_resource and looking
+    for referencedTables in the response.
+    """
+    TIMESTAMP_KEY = 'timestamp'
+    PROJECT_ID_KEY = 'projectid'
+
+    def init(self, conf):
+        credentials = GoogleCredentials.get_application_default()
```

Comment:
I wonder whether we could make the credential specify with configure (example: https://github.com/lyft/amundsendatabuilder/blob/master/databuilder/extractor/hive_table_metadata_extractor.py#L62) which user could specify the credential in the DAG when using the extractor and fallback to default credential if no credential is not provided from user?

Reply:
Yes, this is something to be considered still. The credentials being used can be set with the GOOGLE_APPLICATION_CREDENTIALS environment variable, which then point to a json file on disk:

https://cloud.google.com/docs/authentication/getting-started

So this can be set outside the application. In airflow for example you can have multiple json keyfiles for different accounts if you wanted, but there are also methods for serializing the whole json into a text field:

https://github.com/apache/airflow/blob/master/airflow/contrib/hooks/gcp_api_base_hook.py